app/controllers/api/infrastructures_controller.rb

```diff
@@ -14,12 +14,19 @@ def profile_by_cluster_name
       }, status: :not_found and return
     end
 
+    consul_hosts = @infrastructure.infrastructure_components.
+      where(component_type: 'consul').
+      pluck(:ipaddress).map { |ip| "#{ip}:#{Figaro.env.default_consul_port}" }
+    consul_ipaddresses = @infrastructure.fetch_manifest_ipaddresses('consul')
+    consul_hosts = consul_ipaddresses.empty? ? consul_hosts : consul_ipaddresses
+    consul_host = consul_ipaddresses.empty? ?  @infrastructure.consul_host : consul_hosts.first
+
```

Comment:
i see these block of codes are repeated in 4 files:  app/controllers/api/apps_controller.rb,  app/controllers/api/infrastructures_controller.rb,  app/controllers/api/v2/apps_controller.rb,  app/controllers/api/v2/infrastructures_controller.rb. Can we make it DRY?

Reply:
changed. please check Mas.